components/Article/PayNote.js

```diff
@@ -71,137 +70,210 @@ const memberShipQuery = gql`
+        hasActiveMembership: false,
+        inNativeIOSApp: false
+      },
+      'button'
+    )
+  )
+  .concat(
+    generateNotes(
+      IOS_VARIATIONS,
+      {
+        isEligibleForTrial: false,
+        inNativeIOSApp: true
+      },
+      null
+    )
+  )
+  .concat(
+    generateNotes(
+      THANK_YOU_VARIATIONS,
+      {
+        trialSignup: '1',
+        campaignId: 'any',
+        isEligibleForTrial: false
+      },
+      'trialForm'
+    )
   )
-})
 
-const BuyButton = ({ variation, position }) => {
-  return (
-    <Button
-      primary
-      onClick={trackEventOnClick(
-        ['PayNote', `pledge ${position}`, variation.key],
-        () => goTo('pledge')
-      )}
-    >
-      <Translation
-        baseKey={variation.key}
-        position={position}
-        element='buy/button'
-      />
-    </Button>
+const meetTarget = target => payNote => {
+  const targetKeys = new Set(Object.keys(payNote.target))
+  if (target.trialSignup) targetKeys.add('trialSignup')
+  if (target.campaignId) targetKeys.add('campaignId')
+  return Array.from(targetKeys).every(
+    key =>
+      payNote.target[key] === 'any' ||
+      payNote.target[key] ===
+        (typeof payNote.target[key] === 'boolean' ? !!target[key] : target[key])
   )
 }
 
-const TrialLink = compose(withT)(({ t, variation }) => {
-  const tKey = 'article/payNote/secondaryAction'
-  return (
-    <div {...styles.aside}>
-      {t.elements(`${tKey}/text`, {
-        link: (
-          <a
-            key='trial'
-            href={routes.find(r => r.name === 'trial').toPath()}
-            onClick={trackEventOnClick(
-              ['PayNote', 'preview after', variation],
-              () => goTo('trial')
-            )}
-          >
-            {t(`${tKey}/linkText`)}
-          </a>
-        )
-      })}
-    </div>
+const goTo = route => Router.pushRoute(route).then(() => window.scrollTo(0, 0))
+
+const generateKey = (note, index) => {
+  return { ...note, key: `custom-${index}` }
+}
+
+const disableForIOS = note => {
+  return { ...note, target: { ...note.target, inNativeIOSApp: false } }
+}
+
+const enableForTrialSignup = note => {
+  return note.before.cta === 'trialForm'
+    ? { ...note, target: { ...note.target, trialSignup: 'any' } }
+    : note
+}
+
+const hasCta = cta => note => note.before.cta === cta
+
+const hasTryAndBuyCtas = notes =>
+  notes.some(hasCta('button')) &&
+  notes.some(hasCta('trialForm')) &&
+  notes.every(n => n.before.cta === 'trialForm' || n.before.cta === 'button')
+
+const getPayNote = (target, seed, tryOrBuy, customPayNotes = []) => {
+  const targetedCustomPaynotes = customPayNotes
+    .map(generateKey)
+    .map(disableForIOS)
+    .map(enableForTrialSignup)
+    .filter(meetTarget(target))
+
+  if (targetedCustomPaynotes.length)
+    return getElementFromSeed(targetedCustomPaynotes, seed, MAX_PAYNOTE_SEED)
+
+  const targetedPredefinedNotes = predefinedNotes.filter(meetTarget(target))
+
+  if (targetedPredefinedNotes.length)
```

Comment:
Those nested ifs are the same as:
```js
  if (targetedPredefinedNotes.length && hasTryAndBuyCtas(targetedPredefinedNotes)) {
```
right? did you intent to lead out the curly brackets? (`{}`)

Or did you intent to return `undefined` when `targetedPredefinedNotes.length` is zero?

Maybe do an early return here instead.

```suggestion
  if (!targetedPredefinedNotes.length) {
    return null
  }
```